x/starname/keeper/querier.go

```diff
@@ -314,3 +315,55 @@ func queryBrokerDomains(ctx sdk.Context, keeper *Keeper, broker sdk.AccAddress,
 	}
 	return &types.QueryBrokerDomainsResponse{Domains: domains, Page: page}, nil
 }
+
+//Yield return an estimation of the delegators annualized yield based on the last 100k blocks
+func (q grpcQuerier) Yield(ctx context.Context, _ *types.QueryYieldRequest) (*types.QueryYieldResponse, error) {
+	var response types.QueryYieldResponse
+
+	apy, err := calculateYield(sdk.UnwrapSDKContext(ctx), q.keeper)
+	if err != nil {
+		return nil, err
+	}
+	response.Apy = apy
+	return &response, err
+}
+
+func calculateYield(ctx sdk.Context, keeper *Keeper) (sdk.Dec, error) {
+	// TODO: dont rely on an hardcoded estimate
+	const EstimatedBlockTime = 4
+
+	totalFees, numBlocks := keeper.GetBlockFeesSum(ctx)
+
+	// Compute the reward pool from the collected fees for the last numBlocks blocks
+	communityTax := keeper.DistributionKeeper.GetCommunityTax(ctx)
```

Comment:
Don't deduct the community tax because we're calculating the yield for the chain as a whole.